docu/Concepts/TechnicalRequirements/Architecture/TechnicalOverview.md

```diff
@@ -36,5 +30,76 @@ This document describes the technical overview for the Gradido infrastructur. Be
 
 ![DatabaseModel](../image/GradidoBusinessDBSkeema.png)
 
+# Backend Refactoring (09.2022)
+
+The current implementation of the business logic is concentrated in Resolver classes. Each resolver contains the implementation to handle the data access per typeORM classes to the database including all technical detailed about the explicit transaction handling especially for write access. There is no reuse of code in generalized classes to access for example the user data. Each resolver class, which needs user data has its own implementation of these more or less technical details.
+
+## current Business-Layer
+
+At first the following pictures will show for each current existing resolver class its usage of the backend components. Afterwards an overview picture shows the dependencies between all resolver classes.
+
+### AdminResolver
+
+![img](../image/BE-AdminResolver-IST.png)
+
+### BalanceResolver
+
+![img](../image/BE-BalanceResolver-IST.png)
+
+### CommunityResolver
+
+![img](../image/BE-CommunityResolver-IST.png)
+
+### ContributionMessageResolver
+
+![img](../image/BE-ContributionMessageResolver-IST.png)
+
+### ContributionResolver
+
+![img](../image/BE-ContributionResolver-IST.png)
+
+### GdtResolver
+
+![img](../image/BE-GdtResolver-IST.png)
+
+### KlicktippResolver
+
+![img](../image/BE-KlicktippResolver-IST.png)
+
+### StatisticsResolver
+
+![img](../image/BE-StatisticsResolver-IST.png)
+
+### TransactionLinkResolver
+
+![img](../image/BE-TransactionLinkResolver-IST.png)
+
+### TransactionsResolver
+
+![img](../image/BE-TransactionResolver-IST.png)
+
+### UserResolver
+
+![img](../image/BE-UserResolver-IST.png)
+
+### Overview Resolver Dependencies
+
+![img](../image/BE-Overview-IST.png)
+
+## future Business-Layer
+
+The motivation of this refactoring is to extent the backend with a horizontal data-access-layer containing data-access-objects (postfix DAO). All data access implementations in the resolver classes will be shifted in these new data-access classes. There will be a DAO-class for each data object or in detail for each typeORM class and offer an api for the resolver classes to satify their business requirements. The necessary technical database classes like typeorm will be hidded from resolver access and will only be usable from the new data access layer dao classes.
+
+It is still open, if the data, which is transferred between the resolver classes and the dao classes are from type *graphql model* or from a new business oriented *transfer object model*. In consequence there is always a mapping between the *typeorm* over *transfer object* to *graphql model* during data access throught the different layers necessary.
```

Comment:
I would like to have a clear separation here. In the resolver, we receive database entities from the DAO. The resolver converts them to graphQL objects. The resolver passes the parameters to the DAO via typescript interfaces defined in the DAO

Reply:
If I understand it right we would make "create" methods that would be called with an object to create an object on the database in the resolvers we would have calls like the following:
```JavaScript
const userRepository = getRepository('User')
userRepository.create({gradidoID: .., firstName: .., lastName: .., .....})
```
And in the Repository we would define the object that we expect:
```JavaScript
async create({gradidoID: number, firstName: string, lastName: string, ....}: UserModel) {
  const dbUser = new DbUser()
  dbUser.gradidoID = gradidoID
  dbUser.email = email
  dbUser.firstName = firstName
  dbUser.lastName = lastName
  await queryRunner.manager.save(dbUser)
})
```